MEQP2/Sniffs/Classes/ConstructorOperationsSniff.md

```diff
@@ -0,0 +1,32 @@
+# Rule: Only DI operations are allowed in constructor
+## Background
+In Magento 2, the PHP constructor of a class is used to inject dependencies: Constructor-based Dependency Injection (DI).
+Because the constructor its purpose is to inject dependencies, other operations (or other logic) should not be part of the
+constructor.
+
+## Reasoning
+The constructor should only assign the values of constructor arguments to internal variables, like shown below:
+
+```php
+class Foo
+{
+    private $bar;
+
+    public function __construct(
+        Bar $bar
+    ) {
+        $this->bar = $bar;
+    }
+}
+```
+
+Any other logic than these dependency assignment operators should be excluded from the constructor. Having additional logic in
+the constructor makes it harder to debug and test the class.
+
+## How it works
+This rule checks the constructor code to see if any other operators than `a + b` is used. If so, a warning is generated.
```

Comment:
`+` is allowed? Why? Or did you mean `a = b`?

Reply:
That's why we have reviewing. It was a test. I was busy that day with a lot of JS calculations. Fixed this in the PR. https://github.com/jissereitsma/extdn-phpcs/commit/9e9bb1d805aa607a80d9e9a55f6fda2959ea123a